src/commonMain/kotlin/com/hoc081098/flowext/eager.kt

```diff
@@ -0,0 +1,149 @@
+}
+
+/**
+ * Transforms elements emitted by the original flow by applying [transform], that returns another flow,
+ * and then merging and flattening these flows.
+ *
+ * This operator calls [transform] *sequentially* and then concatenates the resulting flows with a [concurrency]
+ * limit on the number of concurrently collected flows.
+ * It is a shortcut for `map(transform).flattenConcatEager(concurrency)`.
+ * See [flattenConcatEager] for details.
+ *
+ * ### Operator fusion
+ *
+ * Applications of [flowOn], [buffer], and [produceIn] _after_ this operator are fused with
+ * its concurrent merging so that only one properly configured channel is used for execution of merging logic.
+ *
+ * @param concurrency controls the number of in-flight flows, at most [concurrency] flows are collected
+ * at the same time. By default, it is equal to [DEFAULT_CONCURRENCY].
+ */
+@FlowPreview
+public fun <T, R> Flow<T>.flatMapConcatEager(
+  concurrency: Int = DEFAULT_CONCURRENCY,
+  transform: suspend (value: T) -> Flow<R>,
+): Flow<R> = map(transform).flattenConcatEager(concurrency)
+
+/**
+ * Flattens the given flow of flows into a single flow in a sequential manner, without interleaving nested flows.
+ * But unlike [flattenConcat] collecting nested flows performed concurrently with a given [concurrency] limit
+ * on the number of concurrently collected flows.
+ *
+ * If [concurrency] is more than 1, then inner flows are collected by this operator *concurrently*.
+ * With `concurrency == 1` this operator is identical to [flattenConcat].
+ *
+ * ### Operator fusion
+ *
+ * Applications of [flowOn], [buffer], and [produceIn] _after_ this operator are fused with
+ * its concurrent merging so that only one properly configured channel is used for execution of merging logic.
+ *
+ * When [concurrency] is greater than 1, this operator is [buffered][buffer] by default
+ * and size of its output buffer can be changed by applying subsequent [buffer] operator.
+ *
+ * @param concurrency controls the number of in-flight flows, at most [concurrency] flows are collected
+ * at the same time. By default, it is equal to [DEFAULT_CONCURRENCY].
+ */
+@FlowPreview
+public fun <T> Flow<Flow<T>>.flattenConcatEager(concurrency: Int = DEFAULT_CONCURRENCY): Flow<T> {
+  require(concurrency > 0) { "Expected positive concurrency level, but had $concurrency" }
+  return if (concurrency == 1) flattenConcat() else flattenConcatEagerInternal(this, concurrency)
+}
+
+private fun <T> flattenConcatEagerInternal(flow: Flow<Flow<T>>, concurrency: Int) = channelFlow {
+  val collectors = Channel<ProxyCollector<T>>(concurrency - 1)
+  val job: Job? = coroutineContext[Job]
+  launch {
+    job?.ensureActive()
+    for (collector in collectors) {
+      collector.collectTo(this@channelFlow)
+    }
+  }
+  flow
+    .onCompletion { collectors.close() }
+    .collect { inner ->
+      job?.ensureActive()
+      val collector = ProxyCollector(Channel<T>(Channel.BUFFERED, BufferOverflow.SUSPEND))
```

Comment:
```suggestion
      val collector = ProxyCollector(Channel<T>())
```
I think `collector.collectTo(this@channelFlow)` and `this@channelFlow.send` will control the backpressure, so let `collector` be a `rendezvous channel`

Reply:
well, I think that's not how this operator should work. It should collect all values even on not currently connected to output collectors. For example you have:
```kotlin
flowOf(1, 2)
    .flatMapConcatEager { value ->
        flowOf {
            doSomeWorkForFirstEmit()
            emit(...)
            doSomeWorkForSecondEmit()
            emit(...)      
        }
    }
    .collect()
```
If we choose `rendezvous channel` then `doSomeWorkForSecondEmit()` for `value == 2` will wait until flow for `value == 1` will be fully executed because of that backpressure. I tested this case with rxjava3 Observable and it actually doesn't wait, they executed parrallel despite of need to being buffered in some way. Also I found out that in rx there is [`bufferSize`](https://github.com/ReactiveX/RxJava/blob/3.x/src/main/java/io/reactivex/rxjava3/core/Observable.java#L7061) parameter that hints size of each nested Observable, so I think I will extract channel capacity as parameter too

Reply:
I think it is good 💯